main.cpp

```diff
@@ -0,0 +1,255 @@
+#include <iostream>
+#include <climits>
+
+
+template <typename T>
+class Vector {
+private:
+    T* data;
+    std::size_t capacity{};
+    std::size_t size{};
+
+public:
+    Vector( ) : capacity{0}, size{0}, data{nullptr} {}
+
+    Vector( std::size_t size_ ) : capacity{ size_ * 2 }, size{ size_ }
+    {
+        data = new T[ capacity ];
+    }
+
+    Vector(const std::initializer_list<T>& listOfElements) : capacity(listOfElements.size() * 2), size(listOfElements.size()) {
+        data = new T[ capacity ];
+        std::size_t i = 0;
+        for (const T& element : listOfElements) {
+            data[i++] = element;
+        }
+    }
+
+    Vector( std::size_t size_, const T& init_element ) : capacity( size_ * 2 ), size( size_)
+    {
+        data = new T[ capacity ];
+        for( std::size_t i = 0; i < size; ++i)
+        {
+            data[i] = init_element;
+        }
+    }
+
+    Vector( const Vector<T>& other )
+    {
+        capacity = other.capacity;
+        size = other.size;
+        data = new T[capacity];
+
+        for( std::size_t i = 0; i < this->size; ++i ) {
+            data[i] = other.data[i];
+        }
+    }
+
+    Vector( Vector<T>&& other ) noexcept
+    {
+        capacity = other.capacity;
+        size = other.size;
+        data = other.data;
+        other.data = nullptr;
+        other.size = 0;
+        other.capacity = 0;
+    }
+
+    Vector<T>& operator=( const Vector<T>& other )
+    {
+        if( this!= &other ) {
+            capacity = other.capacity;
+            size = other.size;
+            data = new T[capacity];
```

Comment:
`data` might contain pointer to allocated memory, this is a potential memory leak.
`delete[] data` before assigning anything to it.